### ISSUE TITLE
Support for empty log segments

### ISSUE THREAD

**Issue by aivinog1** [Zelldon/zdb#309]
As far as I understand modern Zeebe log code we could have a situation where there exists a current segment with data and there could be the next empty log segment initialized with zeros (if you choose to preallocate segments). But `zdb` currently cannot open a partition log with such segments:
```
aivinog1@MacBook-Pro Downloads % zdb log status -p raft-partition/partitions/1

java.lang.IllegalStateException: Expected version byte to be one [1 3] but got 0 instead.
	at io.zell.zdb.journal.file.SegmentDescriptor.getEncodingLengthForVersion(SegmentDescriptor.java:235)
	at io.zell.zdb.journal.file.SegmentsManager.readDescriptor(SegmentsManager.java:165)
	at io.zell.zdb.journal.file.SegmentsManager.loadExistingSegment(SegmentsManager.java:224)
	at io.zell.zdb.journal.file.SegmentsManager.loadSegments(SegmentsManager.java:144)
	at io.zell.zdb.journal.file.SegmentsManager.open(SegmentsManager.java:112)
	at io.zell.zdb.journal.file.SegmentedReadOnlyJournal.<init>(SegmentedReadOnlyJournal.java:42)
	at io.zell.zdb.journal.file.SegmentedJournalBuilder.build(SegmentedJournalBuilder.java:107)
	at io.zell.zdb.log.LogFactory$Companion.newReader(LogFactory.kt:44)
	at io.zell.zdb.log.LogStatus.<init>(LogStatus.kt:24)
	at io.zell.zdb.LogCommand.status(LogCommand.java:47)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2066)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at io.zell.zdb.ZeebeDebugger.main(ZeebeDebugger.java:60)
aivinog1@MacBook-Pro Downloads % cd raft-partition/partitions/1
aivinog1@MacBook-Pro 1 % hexdump -C -n 100 raft-partition-partition-1-124.log 
00000000  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
00000060  00 00 00 00                                       |....|
00000064
aivinog1@MacBook-Pro-Alfa 1 % hexdump -C -n 100 raft-partition-partition-1-123.log
00000000  02 08 00 04 00 07 00 01  00 e2 98 6b 2f 00 00 00  |...........k/...|
00000010  00 14 00 03 00 07 00 01  00 7b 00 00 00 00 00 00  |.........{......|
00000020  00 24 dd 87 00 00 00 00  00 00 00 00 08 01 0c 00  |.$..............|
00000030  01 00 07 00 01 00 0b fd  1c 2e 00 00 00 00 59 44  |..............YD|
00000040  00 00 10 00 02 00 07 00  01 00 24 dd 87 00 00 00  |..........$.....|
00000050  00 00 eb c1 ac 01 00 00  00 00 3d 44 00 00 09 00  |..........=D....|
00000060  01 00 08 00                                       |....|
00000064
aivinog1@MacBook-Pro 1 % ls -alt
total 524312
-rw-r--r--@  1 aivinog1  staff  134217728 11 окт 21:39 raft-partition-partition-1-123.log
-rw-r--r--@  1 aivinog1  staff        256 11 окт 21:39 raft-partition-partition-1.meta
drwxr-xr-x@  2 aivinog1  staff         64 11 окт 21:30 pending
drwxr-xr-x@ 12 aivinog1  staff        384 11 окт 21:30 runtime
drwxr-xr-x@  4 aivinog1  staff        128 11 окт 21:30 snapshots
drwxr-xr-x@ 10 aivinog1  staff        320 11 окт 18:00 .
-rw-r--r--@  1 aivinog1  staff  134217728 11 окт 17:57 raft-partition-partition-1-124.log
drwxr-xr-x@  5 aivinog1  staff        160 11 окт 14:04 ..
-rw-r--r--@  1 aivinog1  staff          1 11 окт 14:04 .raft-partition-partition-1.lock
-rw-r--r--@  1 aivinog1  staff         78 11 окт 14:04 raft-partition-partition-1.conf
aivinog1@MacBook-Pro 1 % 
``` 

If you think that Zeebe can not have an empty segment in the runtime - also please let me know 🙂

**Comment by Zelldon** [comment]
Hey @aivinog1 I think the problem is more the version check which was removed with 8.3 again. I will fix it soon, since we also need that for our incident response

**Comment by Zelldon** [comment]
BTW Thanks for reporting and trying zdb!

**Comment by Zelldon** [comment]
Fixed by #311
